os/Kconfig

```diff
@@ -198,6 +198,29 @@ config BUILD_KERNEL
 
 endchoice # Build configuration
 
+
+config APP_BINARY_SEPARATION
+	bool "Application binary separation build"
+	depends on BUILD_PROTECTED
+	default n
+	---help---
+		Enables application binary separation.
+		This config is tightly coupled to CONFIG_BUILD_PROTECTED. When protected
+		build is selected, then selecting this config generates separate binary for
+		each user space application, apart from a separate kernel binary. Selecting
+		config also applies MPU based memory isolation and protection for each app.
+
+if APP_BINARY_SEPARATION
+
+config NUM_APPS
```

Comment:
Should we add max limit for NUM_APPS to memory regions?

Reply:
I think its the developer's responsibility to set the number of apps based on the available RAM. Also, it is not related to the MPU regions support. We can support any number of apps using a small number of MPU regions.

Reply:
Ok, how about updating the comment properly with limitation?

Reply:
Ok. done.